### PR TITLE
chore(ci): set up ci

### PR DESCRIPTION
**Pull request by kurtisvg**
Sets up basic CI for the repo, including:
- CODEOWNERS
- header-checker-lint
- labels
- sync-repo-settings
- golangci-lint
- tests on pr/merge

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -17,5 +17,5 @@ package main
 import "fmt"
 
 func main() {
-    fmt.Println("hello world")
-}
\ No newline at end of file
+	fmt.Println("hello world")
```
Comment (Yuan325) [inline main.go:20]: Wondering if this is on purpose?

File: .github/sync-repo-settings.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+branchProtectionRules:
+  - pattern: main
+    isAdminEnforced: true
+    requiredStatusCheckContexts:
+      - "cla/google"
+      - "lint"
+      - "conventionalcommits.org"
+      - "header-check"
+    # - Add required status checks like presubmit tests
+      - integration tests (ubuntu-latest)
```
Comment (averikitsch) [inline .github/sync-repo-settings.yaml]: This may need quotes

File: .github/workflows/lint.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+# Declare default permissions as read only.
+permissions: read-all
+
+jobs:
+  lint:
+    if: "${{ github.event.action != 'labeled' || github.event.label.name == 'tests: run' }}"
+    name: run lint
+    runs-on: ubuntu-latest
+    permissions:
```
Comment (averikitsch) [inline .github/workflows/lint.yaml:32]: Do you need these permissions if you set read-all above?

File: .github/workflows/lint.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    types: [labeled]
+
+# Declare default permissions as read only.
+permissions: read-all
+
+jobs:
+  lint:
+    if: "${{ github.event.action != 'labeled' || github.event.label.name == 'tests: run' }}"
+    name: run lint
+    runs-on: ubuntu-latest
```
Comment (averikitsch) [inline .github/workflows/lint.yaml:28]:
```suggestion
    runs-on: ubuntu-latest
    concurrency:
      group: ${{ github.workflow }}-${{ github.ref }}
      cancel-in-progress: true
```
Best practice to cancel old runs

File: main.go (hunk 1 above)
Comment (kurtisvg) [reply main.go:20]: Yes -- I seemed to have used spaces and goimports fixed it for me

File: .github/workflows/sync-labels.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+permissions: read-all
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
```
Comment (averikitsch) [inline .github/workflows/sync-labels.yaml:31]: Do you need to checkout?

Comment (kurtisvg) [reply .github/workflows/sync-labels.yaml:31]: Looks like it: https://github.com/micnncim/action-label-syncer?tab=readme-ov-file#create-workflow

**Pull request merged**